### PR TITLE
Bump versions of H2, Netty, Slf4J and Commons-codec

### PR DESCRIPTION
**Pull request by hylkevds**
For your convenience, here is one PR to fix 4 versions.
The tricky one was H2, since the `DataType` interface changed. On the plus side, that interface now has a nice generic, and an abstract default implementation, so it does clean things up a bit.

### CODE REVIEW

File: broker/pom.xml [hunk 1]
```diff
@@ -14,26 +14,26 @@
...
-        <netty.version>4.1.73.Final</netty.version>
+        <netty.version>4.1.77.Final</netty.version>
         <!-- Check Netty pom.xm to know the right version of tcnative, for example
-             https://github.com/netty/netty/blob/netty-4.1.73.Final/pom.xml#L545 -->
-        <netty.tcnative.version>2.0.46.Final</netty.tcnative.version>
+             https://github.com/netty/netty/blob/netty-4.1.73.Final/pom.xml#L531 -->
+        <netty.tcnative.version>2.0.52.Final</netty.tcnative.version>
         <paho.version>1.2.5</paho.version>
-        <h2.version>1.4.199</h2.version>
+        <h2.version>2.1.212</h2.version>
```
Comment (andsel) [inline broker/pom.xml:22]:
```suggestion
        <h2.version>2.1.212</h2.version>
        <slf4j.version>1.7.36</slf4j.version>
```

File: broker/pom.xml [hunk 2]
```diff
@@ -14,26 +14,26 @@
...
     </properties>
 
     <dependencies>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
-            <version>1.7.35</version>
+            <version>1.7.36</version>
```
Comment (andsel) [inline broker/pom.xml]:
```suggestion
            <version>${slf4j.version}</version>
```

File: broker/pom.xml [hunk 3]
```diff
@@ -14,26 +14,26 @@
...
         </dependency>
 
         <dependency>
             <!-- This is used to use log4j12 implementation -->
             <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-log4j12</artifactId>
-            <version>1.7.35</version>
+            <artifactId>slf4j-reload4j</artifactId>
+            <version>1.7.36</version>
```
Comment (andsel) [inline broker/pom.xml]:
```suggestion
            <version>${slf4j.version}</version>
```

**Review by andsel — APPROVED** [review]
LGTM
just little nitpick to parameterize the `slf4j` version because it's used in 2 places

File: broker/pom.xml (hunk 3 above)
Comment (hylkevds) [reply broker/pom.xml]: Good point.
Slf4J is also used in the distribution sub-project, so I'll pull the variable to the parent project.

**Review by andsel — APPROVED** [review]
LGTM

**Comment by hylkevds** [comment]
Missed one "reload", fixed now

**Comment by andsel** [comment]
Thank's @hylkevds  for this fix!

**Pull request merged**